habitat/core/embodied_task.py

```diff
@@ -239,6 +239,45 @@ def __repr__(self):
         )
 
 
+class SequenceSpace(Space):
+    """
+    A Space that describes sequence of other Space. Used to describe list of
+    tokens ids, vectors and etc.
+
+    .. code:: py
+
+        observation_space = SequenceSpace(spaces.Discrete(
+            dataset.question_vocab.get_size()))
+    """
+
+    def __init__(self, space, min_seq_length=0, max_seq_length=1 << 15):
+        self.min_seq_length = min_seq_length
+        self.max_seq_length = max_seq_length
+        self.space = space
+        self.length_select = gym.spaces.Discrete(
+            max_seq_length - min_seq_length
+        )
+
+    def sample(self):
+        seq_length = self.length_select.sample() + self.min_seq_length
+        return [self.space.sample() for _ in range(seq_length)]
+
+    def contains(self, x):
+        if not isinstance(x, Sized):
+            return False
+
+        if self.min_seq_length <= len(x) <= self.min_seq_length:
```

Comment:
I think this should be `self.min_seq_length <= len(x) <= self.max_seq_length`, or maybe something else, I am slightly lost on the purpose of this function.